### ISSUE TITLE
Standardize tasks for all build types

### ISSUE THREAD

**Issue by rjanja** [labzero/bootleg#291]
Currently all tasks are loaded, so we have a `:docker_build`, `:local_build` and `:remote_build` because they need to have distinct names.

Although we have a generic `:build` task, all it does is invoke a single task from the above three based on the build type.

We should consider adding other generic build-related tasks:
- verify_config
- compile
- generate_release
- copy_release

Some of these task names are already in use, so this would be a breaking change, but would allow for less confusion when hooking custom tasks.

**Comment by spinlock99** [comment]
Definitely. I noticed this when I was migrating to built-in releases but haven't dug into the code to see where it can be DRYed up. 

Can you take a look at submitting a PR?
